lib/utils.js

```diff
@@ -26,7 +28,91 @@ function queryParameters(queryString) {
   return result;
 }
 
+function isPromise(obj) {
+  return obj instanceof Promise ||
+    !!obj && obj.then && typeof obj.then === 'function';
+}
+
+function isFalsy(val) {
+  return !val ||
+    val instanceof Error ||
+    typeof val === 'string';
+}
+
+// When receives a list of things to wrap as Promises.
+// it will then wait for:
+//  a) all to settle
+//  b) first to reject
+//  c) first to settle with a falsy value
+// c) is sub-optimal, since it has a specific narrow use-case
+// for save hooks
+function when(items) {
+  items = items || [];
+  var resolveAll, rejectAll,
+      allResolvedPromise = new Promise(function(resolve, reject) {
+        resolveAll = resolve;
+        rejectAll = reject;
+      });
+
+  var remaining = 0,
+      settledWith = [],
+      itemsCopy = Array.isArray(items) ?
+        items.slice() :
+        [items];
+
+  var resolveWith = function(data, index) {
+    settledWith[index] = data;
+
+    if (--remaining <= 0) {
+      resolveAll(settledWith);
+    }
+  };
+
+  remaining = itemsCopy.length;
+
+  if (remaining <= 0) {
+    resolveAll();
+    return allResolvedPromise;
+  }
+
+  itemsCopy.forEach(function(item, index) {
+    var promise;
+    if (isPromise(item)) {
+      promise = item;
+    } else if (typeof item === 'function') {
+      var res;
+      try {
+        res = item();
+        if (isPromise(res)) {
+          promise = res;
+        } else {
+          promise = new Promise(function(resolve) {
+            resolve(res);
+          });
+        }
+      } catch(err) {
+        promise = new Promise(function(resolve, reject) {
+          reject(err);
+        });
+      }
+    } else {
+      promise = new Promise(function(resolve, reject) {
+        isFalsy(item) ?
```

Comment:
@svizzari - I'm confused by this - was the intention to check item? as in if someone does `client.on('ticket.save', false)` ?